cmake/SetupCompilers.cmake

```diff
@@ -40,7 +40,9 @@
 # 
 ###############################################################################
 
-set(CMAKE_CXX_STANDARD 14)
+if (NOT RAJA_ENABLE_CLANG_CUDA)
+  set(CMAKE_CXX_STANDARD 14)
```

Comment:
Uhhh...how long have we been doing this? We don't want to be relying on language features that may not be available for our apps. If we allow C++14 features, then we should also have implementation alternatives that do not use anything beyond C++11 (we would probably need to add a #define to enable conditional compilation) until we verify that they work on all platforms we must support